### PR TITLE
manifests: run gateways as non-root user

### LINKED ISSUE #20853: Ability to run ingress/egress gateways as non-root user
**Describe the feature request**
The principle of least privilege suggests that it would be more secure to run the ingress/egress gateway containers as a non-root user. Changes have already been made (see #20311) to run control-plane containers as user 1337, but the gateways are a bit more complicated as they want to bind to ports below 1024, so the envoy process would need to be granted the `CAP_NET_BIND_SERVICE` capability if not running as root.

**Additional context**
@rlenglet gave some useful pointers in [this comment on #20311](https://github.com/istio/istio/pull/20311#issuecomment-581204542).

### PR DESCRIPTION
**Pull request by dgn**
Fixes #20853

### CODE REVIEW

**Comment by linsun** [comment]
LGTM

File: manifests/gateways/istio-egress/templates/deployment.yaml [hunk 1]
```diff
@@ -56,7 +56,9 @@ spec:
             - -c
             - sysctl -w kernel.core_pattern=/var/lib/istio/core.proxy && ulimit -c unlimited
           securityContext:
-            privileged: true
```
Comment (howardjohn) [inline manifests/gateways/istio-egress/templates/deployment.yaml]: this is the core dump, I feel like sysctl may need root? maybe not though, but it's off by default

**Review by howardjohn — COMMENTED** [review]
lgtm in general

File: manifests/gateways/istio-egress/templates/deployment.yaml (hunk 1 above)
Comment (dgn) [reply manifests/gateways/istio-egress/templates/deployment.yaml]: you're right, these are wrong. will change back

**Comment by dgn** [comment]
Closing this for now, this seems to need more work

**Comment by incfly** [comment]
Just curious, whats blocking us from this PR?

**Pull request closed without merging**